tools/core.mk

```diff
@@ -0,0 +1,106 @@
+
+rfcdiff = rfcdiff --width 78 --stdout
+reduction  = $(tools)/clean-for-DTD.xslt
+bap = $(tools)/bap
+bd  = build
+
+draftname = draft-ietf-httpbis
+
+AUTH        = $(draftname)-auth
+CACHE       = $(draftname)-cache
+CONDITIONAL = $(draftname)-conditional
+MESSAGING   = $(draftname)-messaging
+RANGE       = $(draftname)-range
+SEMANTICS   = $(draftname)-semantics
+
+CORE_XML = $(MESSAGING).xml $(SEMANTICS).xml
+TARGETS_ABNF= $(addprefix $(bd)/,$(CORE_XML:.xml=.abnf))
+TARGETS_ABNFAPPENDIX= $(TARGETS_ABNF:.abnf=.abnf-appendix)
+TARGETS_PARSEDABNF= $(TARGETS_ABNF:.abnf=.parsed-abnf)
+
+ABNF =    $(TARGETS_ABNF) \
+          $(TARGETS_PARSEDABNF) \
+          $(TARGETS_ABNFAPPENDIX) \
+          $(bd)/$(MESSAGING).iana-headers \
+          $(bd)/$(SEMANTICS).iana-headers \
+          $(bd)/$(SEMANTICS).iana-methods	\
+          $(bd)/$(SEMANTICS).iana-status-codes \
+          httpbis.abnf
+
+$(bap)/bap:
+	cd $(bap); sh configure
+	cd $(bap); make
+
+.PHONY: abnf
+abnf: $(ABNF)
+
+$(bd)/%.abnf: %.xml $(tools)/extract-artwork.xslt
+	$(saxon) $< $(xreffer) | $(saxon) - $(tools)/extract-artwork.xslt type="abnf2616" >$@
+
+$(bd)/%.parsed-abnf: $(bd)/%.abnf $(bap)/bap
+	$(bap)/bap -i $(bap)/core.abnf < $< | LC_COLLATE=C sort | $(bap)/bap -k -i $(bap)/core.abnf -l 69 >$@
+
+$(bd)/%.abnf-appendix: $(bd)/%.parsed-abnf
+	$(saxon) $*.xml $(bd)/abnf2xml2rfc.xslt abnf="$*.parsed-abnf" >$@
+
+$(bd)/%.xhtml: %.xml $(tools)/rfc2629toXHTML.xslt
+	$(saxon) $< $(tools)/rfc2629toXHTML.xslt > $@
+
+$(bd)/%.iana-headers: %.xml $(tools)/extract-header-defs.xslt
+	$(saxon) $< $(tools)/extract-header-defs.xslt > $@
+
+$(bd)/%.iana-methods: %.xml $(tools)/extract-method-defs.xslt
+	$(saxon) $< $(tools)/extract-method-defs.xslt > $@
+
+$(bd)/%.iana-status-codes: %.xml $(tools)/extract-status-code-defs.xslt
+	$(saxon) $< $(tools)/extract-status-code-defs.xslt > $@
+
+$(bd)/%.iana-warn-codes: %.xml $(tools)/extract-warn-code-defs.xslt
+	$(saxon) $< $(tools)/extract-warn-code-defs.xslt > $@
+
+$(bd)/%.cache-directives: %.xml $(tools)/extract-cache-directives.xslt
+	$(saxon) $< $(tools)/extract-cache-directives.xslt > $@
+
+outlineALL.html: $(TARGETS_XHTML) $(tools)/extractOutline.xslt
+	$(saxon) $(tools)/extractOutline.xslt $(tools)/extractOutline.xslt > $@
+
+httpbis.abnf: $(TARGETS_ABNF)
```

Comment:
Is this in .gitignore?

Reply:
If it's a build product, you will know it changes when the source changes.  Having to review changes with generated output and source is difficult.

Reply:
It is difficult, but it's a guard against unintended ABNF changes.

Reply:
You can use CI for that.